pkg/bgpv1/manager/reconcilerv2/service.go

```diff
@@ -850,6 +871,77 @@ func checkServiceAdvertisement(advert v2alpha1.BGPAdvertisement, advertServiceTy
 	return true, nil
 }
 
+func mergeRoutePolicies(policyA *types.RoutePolicy, policyB *types.RoutePolicy) (*types.RoutePolicy, error) {
+	if policyA == nil || policyB == nil {
+		return nil, fmt.Errorf("route policy is nil")
+	}
+	if policyA.Name != policyB.Name {
+		return nil, fmt.Errorf("route policy names do not match")
+	}
+	if policyA.Type != policyB.Type {
+		return nil, fmt.Errorf("route policy types do not match")
+	}
+
+	communities := sets.NewString()
+	largeCommunities := sets.NewString()
+	mergedPolicy := &types.RoutePolicy{
+		Name:       policyA.Name,
+		Type:       policyA.Type,
+		Statements: []*types.RoutePolicyStatement{},
+	}
+	routePolicyConditionsSeen := sets.NewString()
+
+	// Extract the first policy's statements and communities
+	for _, statement := range policyA.Statements {
+		mergedPolicy.Statements = append(mergedPolicy.Statements, &types.RoutePolicyStatement{
+			Actions:    statement.Actions,
+			Conditions: statement.Conditions,
+		})
+		communities.Insert(statement.Actions.AddCommunities...)
+		largeCommunities.Insert(statement.Actions.AddLargeCommunities...)
+		routePolicyConditionsSeen.Insert(statement.Conditions.String())
+	}
+
+	// Extract the second policy's non-overlapping statements, and communities from all statements
+	for _, statement := range policyB.Statements {
+		if !routePolicyConditionsSeen.Has(statement.Conditions.String()) {
+			mergedPolicy.Statements = append(mergedPolicy.Statements, &types.RoutePolicyStatement{
+				Actions:    statement.Actions,
+				Conditions: statement.Conditions,
+			})
+		}
+		for _, community := range statement.Actions.AddCommunities {
+			communities.Insert(community)
+		}
+		for _, largeCommunity := range statement.Actions.AddLargeCommunities {
+			largeCommunities.Insert(largeCommunity)
+		}
+	}
+
+	bestLocalPreference := int64(0)
```

Comment:
In the original version, I was ignoring Local Pref entirely and just leaving it set to whatever `policyA` had.  Given the complex handling for communities, that didn't feel right.

Here, I'm selecting the highest (best) local preference value across the two policies and using that.  If this seems reasonable, I'll move this into a new function.

Reply:
Thanks, yes it makes sense.